wire-ios-data-model/Source/Repositories/FeatureRepository.swift

```diff
@@ -21,6 +21,8 @@ import Foundation
 // sourcery: AutoMockable
 public protocol FeatureRepositoryInterface {
 
+    var context: NSManagedObjectContext { get }
```

Comment:
Why do we need to expose the context in the protocol?

Reply:
It's ugly, I agree. I was scared of changing the FeatureRepository's implementation.
I used it like this:
```
let isE2EIEnabled = await featureRepository.context.perform {
    featureRepository.fetchE2EI().isEnabled
}
```
But I will have to change the implementation of `fetchE2EI()`, because the initial guard is not concurrency safe:
```
        guard
            let feature = Feature.fetch(name: .e2ei, context: context),
            let featureConfig = feature.config
        else {
            return .init()
        }
```

Reply:
> Why do we need to expose the context in the protocol?

After thinking again, I'd answer your question with yes.
There is a requirement that the `FeatureRepository`'s internal implementation is only called from the same context's queue that was passed into the `FeatureRepository`'s initialiser. 
If that context is not part of the protocol, the requirement cannot be satisfied.